### ISSUE TITLE
[v2.1.1] Inheritance Bug

### ISSUE THREAD

**Issue by talha-asad** [jagi/meteor-astronomy#468]
Schemas:

``` js
// Class Campaign
Campaign = Class.create({
  name: 'Campaign',
  secured: false,
  collection: Campaigns,
  typeField: 'network',
  fields: {
    _id: {type: String, index: 1},
    name: {type: String},
    type: {type: String, optional: true, index: 1},
    remote_id: {type: String, optional: true, index: 1},
    remote_info: {type: [String], default() { return []; }},
  }
});

// Twitter Campaign
TwitterCampaign = Campaign.inherit({
  name: 'TwitterCampaign',
  fields: {
    budget: {
      type: Class.create({
        name: 'CampaignBudget',
        fields: {
          daily: {type: Number},
          total: {type: Number, optional: true}
        }
      }),
      default() { return {}; }
    }
  }
});
```

Client-side with only one campaign published.

``` js
TwitterCampaign.find({}).fetch();
/* [Class {
  _id: "rB6nNvLCqWK4zvxgt",
  name: "Test",
  network: "TwitterCampaign",
  type: undefined,
  remote_id: 'fg6q8',
  remote_info: [],
  budget: Class {
    daily: 15,
    total: 50
  }
}]
*/

Campaign.find({}, {children: true}).fetch();
/* [Class {
  _id: "rB6nNvLCqWK4zvxgt",
  name: "Test",
  network: "TwitterCampaign",
  type: undefined,
  remote_id: 'fg6q8',
  remote_info: [],
  budget: Class {
    daily: undefined,
    total: undefined
  }
}]
*/
```

Notice the difference in nested `budget` field values. Tried playing with `children` property but seems that if you use the parent Class to query, nested Classes don't show actual values.

**Comment by lukejagodzinski** [comment]
I think it was already fixed but not published yet. I will publish a bug fix version probably later today.

**Comment by talha-asad** [comment]
Awesome.

**Comment by THPubs** [comment]
Hi, @jagi is the update live? I'm also having the same issue. :-)

**Comment by lukejagodzinski** [comment]
Released v2.1.2 just now

**Comment by THPubs** [comment]
Cool thanks :-)
